lens-cube/src/main/java/org/apache/lens/cube/parse/UnionCandidate.java

```diff
@@ -21,29 +26,42 @@
   String toStr;
   @Getter
   String alias;
+  CubeQueryContext cubeql;
   /**
    * List of child candidates that will be union-ed
    */
   private List<Candidate> childCandidates;
+  @Getter
+  private QueryAST queryAst;
 
-  public UnionCandidate(List<Candidate> childCandidates, String alias) {
+  public UnionCandidate(List<Candidate> childCandidates, String alias, CubeQueryContext cubeql) {
     this.childCandidates = childCandidates;
     this.alias = alias;
+    this.cubeql = cubeql;
   }
 
   @Override
-  public String toHQL() {
+  public String toHQL() throws LensException {
     return null;
   }
 
   @Override
-  public QueryAST getQueryAst() {
-    return null;
+  public ArrayList<Integer> getAnswerableMeasureIndices() {
+    ArrayList<Integer> mesureIndices = new ArrayList<>();
+    List<StorageCandidate> scs = new ArrayList<StorageCandidate>();
```

Comment:
CandidateUtil.getStorageCandidates() returns a collection . We can directly use it and not create a new ArrayList